core/src/com/unciv/logic/automation/unit/CityLocationTileRanker.kt

```diff
@@ -64,12 +64,32 @@ object CityLocationTileRanker {
         return bestTilesToFoundCity
     }
 
-    @Readonly
-    private fun canSettleTile(tile: Tile, civ: Civilization, nearbyCities: Sequence<City>): Boolean {
+
+  @Readonly
+    private fun canSettleTile(tile: Tile, unit: MapUnit, nearbyCities: Sequence<City>): Boolean {
+        val civ = unit.civ
+        val foundUniques = unit.getMatchingUniques(UniqueType.FoundCity) + unit.getMatchingUniques(UniqueType.FoundPuppetCity)
+        val unique = foundUniques.firstOrNull()!!
+        var canSettleInTileWithUnique = false
+      
+        val uniqueModifier = unique.getModifiers(UniqueType.ConditionalInTiles).firstOrNull()
         val modConstants = civ.gameInfo.ruleset.modOptions.constants
-        if (!tile.isLand || tile.isImpassible()) return false
         if (tile.getOwner() != null && tile.getOwner() != civ) return false
+      
+        if (unique.hasModifier(UniqueType.ConditionalInTiles)) {
+            canSettleInTileWithUnique = !unique.getModifiers(UniqueType.ConditionalInTiles).none{ 
```

Comment:
`none {` - interestingly, https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace shows the rule for lambdas, but not other blocks or fun/if/for... bodies...